File: behavioral/state/pattern/Aufgelegt.java
package jarai.patterns.gof.behavioral.state.pattern;


public class Aufgelegt extends Zustand {
    public Aufgelegt() {

        super("<Hörer ist aufgelegt>");
    }

    public Zustand Abheben() {
        cout << "Hörer wird abgehoben.");
        return new Abgehoben();
    }

    public Zustand AnnehmenAnruf() {
        cout << "Anruf wird angenommen.");
        return new Verbunden();
    }
}

File: behavioral/state/pattern/Aufgelegt.cpp
#include <iostream>

#include "Aufgelegt.h"
#include "Verbunden.h"

Aufgelegt::Aufgelegt()
	: Zustand("<Hörer ist aufgelegt>")
{
}

std::shared_ptr<Zustand> Aufgelegt::Abheben()
{
	std::cout << "Hörer wird abgehoben." << std::endl ;
	return std::make_shared<Abgehoben>();
}

std::shared_ptr<Zustand> Aufgelegt::AnnehmenAnruf()
{
	std::cout << "Anruf wird angenommen." << std::endl ;
	return std::make_shared<Verbunden>();
}
